sort/insertion.go

```
@@ -0,0 +1,22 @@
+//
+// Space Complexity: O(1) as it sorts in-place
+func InsertionSort[T number](arr []T) []T {
+	n := len(arr)
+	for i := range n - 1 {
+		for j := i + 1; j > 0; j-- {
+			if arr[j] < arr[j-1] {
+				arr[j], arr[j-1] = arr[j-1], arr[j]
+			}
+		}
+	}
+	return arr
```

golangci-lint (1.62.2)

return with no blank line before

(nlreturn)